### PR TITLE
Allow expression builder in set operations. Closes #285

### LINKED ISSUE #285: add support for nested `with...select` when using set operations (e.g. `union`).
I've run into a few issues with the union and unionAll functions. I’m using the Postgres dialect.

The ability to wrap the union SQL in parentheses, as using "with" inside of a union will result in a syntax error.

~It would also be great if there could also be a way to provide an array of expressions into the function so we can avoid using a for loop.~

### PR DESCRIPTION
**Pull request by koskimas**

### CODE REVIEW

File: src/query-builder/select-query-builder.ts [hunk 1]
```diff
@@ -1205,8 +1208,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
    *   .union(db.selectFrom('pet').select(['id', 'name']))
    *   .orderBy('name')
    * ```
+   *
+   * You can provide a callback to get an expression builder:
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1212]:
```suggestion
   * You can provide a callback to get an expression builder.
   * In the following example, this allows us to wrap the query in parentheses:
```

File: src/query-builder/select-query-builder.ts [hunk 2]
```diff
@@ -1229,8 +1245,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
    *   .unionAll(db.selectFrom('pet').select(['id', 'name']))
    *   .orderBy('name')
    * ```
+   *
+   * You can provide a callback to get an expression builder:
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1249]:
```suggestion
   * You can provide a callback to get an expression builder.
   * In the following example, this allows us to wrap the query in parentheses:
```

File: src/query-builder/select-query-builder.ts [hunk 3]
```diff
@@ -1253,8 +1282,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
    *   .intersect(db.selectFrom('pet').select(['id', 'name']))
    *   .orderBy('name')
    * ```
+   *
+   * You can provide a callback to get an expression builder:
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1286]:
```suggestion
   * You can provide a callback to get an expression builder.
   * In the following example, this allows us to wrap the query in parentheses:
```

File: src/query-builder/select-query-builder.ts [hunk 4]
```diff
@@ -1301,8 +1356,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
    *   .except(db.selectFrom('pet').select(['id', 'name']))
    *   .orderBy('name')
    * ```
+   *
+   * You can provide a callback to get an expression builder:
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1360]:
```suggestion
   * You can provide a callback to get an expression builder.
   * In the following example, this allows us to wrap the query in parentheses:
```

File: src/query-builder/select-query-builder.ts [hunk 5]
```diff
@@ -1325,8 +1393,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
    *   .exceptAll(db.selectFrom('pet').select(['id', 'name']))
    *   .orderBy('name')
    * ```
+   *
+   * You can provide a callback to get an expression builder:
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1397]:
```suggestion
   * You can provide a callback to get an expression builder.
   * In the following example, this allows us to wrap the query in parentheses:
```

File: src/query-builder/select-query-builder.ts [hunk 6]
```diff
@@ -1205,8 +1208,21 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
...
+   *
+   * ```ts
+   * db.selectFrom('person')
+   *   .select(['id', 'first_name as name'])
+   *   .union((eb) => eb.parens(
+   *     db.selectFrom('pet').select(['id', 'name'])
```
Comment (igalklebanov) [inline src/query-builder/select-query-builder.ts:1218]: should this be `eb.selectFrom` and not `db.selectFrom`?

**Review by igalklebanov — APPROVED** [review]
LGTM! 🚀 

Left some documentation suggestions/comments..

File: src/query-builder/select-query-builder.ts (hunk 6 above)
Comment (koskimas) [reply src/query-builder/select-query-builder.ts:1219]: Yep, good catch!

**Pull request merged**